src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -113,8 +119,9 @@ await bindingContext.ValueProvider.ContainsPrefixAsync(bindingContext.ModelName)
             foreach (var propertyMetadata in context.PropertyMetadata)
             {
                 // This check will skip properties which are marked explicitly using a non value binder.
-                if (propertyMetadata.BinderMetadata == null ||
-                    propertyMetadata.BinderMetadata is IValueProviderMetadata)
+                var bindingSource = GetBindingSource(propertyMetadata.BinderMetadata);
+                if (bindingSource == null ||
+                    bindingSource.IsValueProvider)
```

Comment:
still reading through the PR... but the property name has started to make more sense than the suggested ```IsGreedy```